lib/src/json_schema/json_schema.dart

```diff
@@ -1921,4 +1920,13 @@ class JsonSchema {
   /// Validate, calculate and set the value of the 'required' JSON Schema keyword.
   _setRequiredV6(dynamic value) =>
       _requiredProperties = (TypeValidators.list('required', value))?.map((value) => value as String)?.toList();
+
+  /// Mixin a another JsonSchema into this one. All references must be resolved before calling.
+  mixinForRef(JsonSchema ref) {
+    this._schemaMap.remove(r'$ref');
```

Comment:
What if we `$ref` the same type N times? Do we need to live it around, potentially?

Reply:
We inline the `$ref` right here, so no reason to keep it around. If this `JsonSchema` is used again, we also won't need to merge the schema back it.

Reply:
:+1: this isn't doing quite what I thought it was at first